### ISSUE TITLE
Make lock files for certificate renewal global across swarms rather than local to a swarm

### ISSUE THREAD

**Issue by cpswan** [atsign-foundation/at_server#789]
**Is your feature request related to a problem? Please describe.**

At present we create a lock file in `/var/lock/renew_certs.lock` to stop multiple instances of `create_ren.sh` running over the top of each other. But this still allows multiple instances across our swarms, where we potentially have many scripts at once trying to create certs (and potentially exceeding the rate limits for our CAs)

**Describe the solution you'd like**

Put the lock file on /gluster/@ somewhere so that we don't have multiple scripts running at once. We will also have to consider script scheduling to ensure that each swarm gets its turn when the lock is freed.

**Describe alternatives you've considered**

We could move cert renewal away from the swarms to some external scheduler.

**Additional context**

Issues with Google certs revealed significant clumps of Google issued certs, which likely came about from exceeding rate limits by running renewals over each other.

**Comment by kumarnarendra701** [comment]
Moved this ticket to the next sprint because didn’t get the bandwidth to work on this ticket.

**Comment by kumarnarendra701** [comment]
Moved this ticket to the next sprint because didn’t get the bandwidth to work on this.

**Comment by kumarnarendra701** [comment]
@cpswan - Please review PR and let me know if you see any enhancement.
https://github.com/atsign-company/secondaries-scripts/pull/53

**Comment by cpswan** [comment]
The global lock is now in place, thanks @kumarnarendra701 

But we have some side effects to deal with, per my comments at https://github.com/atsign-company/secondaries-scripts/pull/53#issuecomment-1193792351

- [x] The cron times should be staggered, so each cluster doesn't try to grab the lock at the same time.
- [x] The cron jobs should run much more frequently than daily (perhaps hourly) to give them a chance to grab the lock once it has been released.
- [x] The script run by cron shouldn't do anything if it's already run that day, so its first job should be to check for a log file and if there's one there from that day then exit without ever taking the lock.
- [x] Finding that another script has the lock should be normal, so we don't need an alert in chat for that. But we probably should have an alert if a renewal script hasn't been able to run for more than 3 days.

So moved to PR42 and 3 SP added to cover the extra work.

**Comment by kumarnarendra701** [comment]
@cpswan - As suggested added the above logic on the cert renewal process. Please review and share your feedback.
https://github.com/atsign-company/secondaries-scripts/pull/55

**Comment by kumarnarendra701** [comment]
@cpswan - Script deployed on Staging and configured cron to run every hour. I'll monitor for today and then deploy it to prod.

**Comment by cpswan** [comment]
Ok. Any thoughts @kumarnarendra701 on how we coordinate between staging and production, as they're on different NFS?

**Comment by kumarnarendra701** [comment]
@cpswan - Created script to put lock file in GCS bucket and, as suggested working on granting GCS permission from the role for VM and testing script.

**Comment by cpswan** [comment]
@kumarnarendra701 if the GCS thing is slowing us down then let's move quicker on getting a new lock approach in place for the prod swarms.

**Comment by cpswan** [comment]
Moving to PR43 as we're not done yet.

**Comment by kumarnarendra701** [comment]
@cpswan - Last version script deployed to prod and configured cron to run hourly.

**Comment by cpswan** [comment]
@kumarnarendra701 can you please run a report across the prod swarms to show how many certs are due for renewal in <21 days so we have a measure of any problems we might be facing over previously delayed renewal.

**Comment by kumarnarendra701** [comment]
@cpswan - Please find below the report.

swarm0001 - No certs renewal pending <21 days
swarm0002 - No certs renewal pending <21 days
swarm0003 - No certs renewal pending <21 days
swarm0004 - No certs renewal pending <21 days
canary - No certs renewal pending <21 days

**Comment by kumarnarendra701** [comment]
@cpswan - 
Reg: - Add storage roles in IAM service account that is associated with VM. 
We need to create a new service account and add a storage role and attach it to VM but it requires a reboot. So I have created a service account in the Development GCP project and activated that service account in staging and prod swarms servers to access the GCS bucket.

I have tested the script and raised PR for your review.

PR - https://github.com/atsign-company/secondaries-scripts/pull/56
